patroni/postgresql/slots.py

```diff
@@ -371,6 +378,16 @@ def _ensure_physical_slots(self, slots: Dict[str, Any]) -> None:
                 except Exception:
                     logger.exception("Failed to create physical replication slot '%s'", name)
                 self._schedule_load_slots = True
+            elif not self._postgresql.is_primary() and self._postgresql.major_version >= 110000\
+                    and self._replication_slots[name]['type'] == 'physical':
+                value['restart_lsn'] = self._replication_slots[name]['restart_lsn']
+                lsn = parse_int(value.get('lsn'))
+                if lsn and lsn > value['restart_lsn']:  # The slot has feedback in DCS and needs to be advanced
+                    try:
+                        lsn = format_lsn(lsn)
+                        self._query("SELECT pg_catalog.pg_replication_slot_advance(%s, %s)", name, lsn)
+                    except Exception as exc:
+                        logger.error("Error while advancing replication slot %s to position '%s': %r", name, lsn, exc)
```

Comment:
Would `logger.exception` or `logger.error(..., exc_info=True)` work here?

Reply:
Ah, I see now this is done throughout the code. I expect there's good reason for that...

Reply:
There is no real good reason to show a full trace in this case.